sync_diff_inspector/diff.go

```diff
@@ -123,13 +143,125 @@ func (df *Diff) CreateDBConn(cfg *Config) (err error) {
 	return nil
 }
 
+func (df *Diff) adjustTableConfigBySubTask(cfg *Config) (err error) {
+	allTablesMap, err := df.GetAllTables(cfg)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	baLists := make(map[string]*filter.Filter)
+	tableRouters := make(map[string]*router.Table)
+
+	for _, subTaskCfg := range df.subTaskCfgs {
+		baList, err := filter.New(subTaskCfg.CaseSensitive, subTaskCfg.BAList)
+		if err != nil {
+			return err
+		}
+		baLists[subTaskCfg.SourceID] = baList
+
+		tableRouter, err := router.NewTableRouter(subTaskCfg.CaseSensitive, []*router.TableRule{})
+		if err != nil {
+			return err
+		}
+		for _, rule := range subTaskCfg.RouteRules {
+			err := tableRouter.AddRule(rule)
+			if err != nil {
+				return err
+			}
+		}
+		tableRouters[subTaskCfg.SourceID] = tableRouter
+	}
+
+	// get all source table's matched target table
+	// target database name => target table name => all matched source table instance
+	sourceTablesMap := make(map[string]map[string][]TableInstance)
+	for instanceID, allSchemas := range allTablesMap {
+		if instanceID == df.targetDB.InstanceID {
+			continue
+		}
+
+		for schema, allTables := range allSchemas {
+			if filter.IsSystemSchema(schema) {
+				continue
+			}
+
+			for table := range allTables {
+				if baLists[instanceID] != nil {
+					tbs := []*filter.Table{{Schema: schema, Name: table}}
+					tbs = baLists[instanceID].ApplyOn(tbs)
+					if len(tbs) == 0 {
+						continue
+					}
+				}
+
+				targetSchema := schema
+				targetTable := table
+				if tableRouters[instanceID] != nil {
+					targetSchema, targetTable, err = tableRouters[instanceID].Route(schema, table)
+					if err != nil {
+						return errors.Errorf("get route result for %s.%s.%s failed, error %v", instanceID, schema, table, err)
+					}
+				}
+
+				if _, ok := sourceTablesMap[targetSchema]; !ok {
+					sourceTablesMap[targetSchema] = make(map[string][]TableInstance)
+				}
+
+				if _, ok := sourceTablesMap[targetSchema][targetTable]; !ok {
+					sourceTablesMap[targetSchema][targetTable] = make([]TableInstance, 0, 1)
+				}
+
+				sourceTablesMap[targetSchema][targetTable] = append(sourceTablesMap[targetSchema][targetTable], TableInstance{
+					InstanceID: instanceID,
+					Schema:     schema,
+					Table:      table,
+				})
+			}
+		}
+	}
+
+	for schema, tables := range sourceTablesMap {
+		for table, sourceTables := range tables {
+			if sourceTables == nil || len(sourceTables) == 0 {
+				continue
+			}
+
+			// TODO: support sql-mode
```

Comment:
seems there's only `EnableANSIQuotes` in `SubTaskConfig`

Reply:
support it in 6ade6b5